### PR TITLE
Gce minimum report

### PR DESCRIPTION
**Pull request by confunguido**
Provides a skeleton for a report. Also includes changes from the pop loader so probably need to land that first.

### CODE REVIEW

**Review by ChiragKumar9 — CHANGES_REQUESTED** [review]
Thanks, @confunguido. I have one big comment on the need to use `add_periodic_plan_with_phase` here to ensure the report is consistently reporting out at the same time. The rest of my comments are mainly stylistic. I did not review the `population_loader` in this PR since that is the subject elsewhere.

File: src/population_loader.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+define_person_property_with_default!(Alive, bool, true);
+
+define_person_property!(CensusTract, usize);
+
+fn create_person_from_record(
+    context: &mut Context,
+    person_record: &PeopleRecord,
+) -> Result<PersonId, IxaError> {
+    let tract: usize = String::from_utf8(person_record.homeId[..11].to_owned())
+        .expect("Home id should have 11 digits for tract + home id")
```
Comment (KOVALW) [inline src/population_loader.rs:31]: Home ID should have 15 digits, the census tract itself is 11 digits and home ID subcomponent is four.

File: src/periodic_report_population.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+pub fn init(context: &mut Context, output_dir: &Path) -> Result<(), IxaError> {
+    let parameters = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .clone();
+
+    context
+        .report_options()
+        .overwrite(true)
```
[the remark this answers is not in the document]
Comment (ChiragKumar9) [reply src/periodic_report_population.rs:77]: nevermind, we should use command line args instead. i can do tha

File: src/periodic_report_population.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    PopulationReportData,
+    PopulationReportData {
+        census_tract_set: HashSet::new(),
+    }
+);
+
+create_report_trait!(PersonReportItem);
+
+fn send_population_report(context: &mut Context, report_period: f64) {
+    let population_data = context.get_data_container_mut(PopulationReportPlugin);
```
[the remark this answers is not in the document]
Comment (confunguido) [reply src/periodic_report_population.rs:41]: Done

File: src/periodic_report_population.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    let current_census_set = population_data.census_tract_set.clone();
+    for age_it in 0..100 {
```
[the remark this answers is not in the document]
Comment (confunguido) [reply src/periodic_report_population.rs:44]: I wouldn't be opposed to hard-coding age as MAXAGE or something like that as a static variable. Not sure that it needs to be in the input, but yeah we can discuss for next iteration.

File: src/population_loader.rs (hunk 1 above)
Comment (confunguido) [reply src/population_loader.rs:31]: I would let this to be handled in the population skeleton branch

**Comment by confunguido** [comment]
Closing to reopen in population skeleton branch

**Pull request closed without merging**